solver/llbsolver/ops/exec.go

```diff
@@ -483,15 +477,23 @@ func (sm *secretMountInstance) Mount() ([]mount.Mount, error) {
 	}
 
 	if err := mount.All([]mount.Mount{tmpMount}, dir); err != nil {
-		return nil, errors.Wrap(err, "unable to setup secret mount")
+		cleanupDir()
+		return nil, nil, errors.Wrap(err, "unable to setup secret mount")
 	}
 	sm.root = dir
 
+	cleanup := func() error {
+		if err := mount.Unmount(dir, 0); err != nil {
```

Comment:
the original version had a condition `if sm.root != ""` https://github.com/moby/buildkit/pull/1137/files#diff-0d2115e5c32fba316a7eaa987331881cL528

Reply:
and in here we are guaranteed that this is the case